FrontEnd/src/ProfilePage/EditingProfile.jsx

```diff
@@ -0,0 +1,101 @@
+import { useState, useEffect } from "react";
+import { doc, updateDoc } from "firebase/firestore";
+import { database } from "../UserAuthentication/FirebaseConfig";
+import  fetchCategoryNames  from "../Categories";
+import Modal from "../Modal";
+
+const EditProfile = ({ userUID, userData,isEditProfileModalShown, setIsEditProfileModalShown }) =>{
+  const DATABASE_FOLDER_NAME = "users";
+  const userCollectionRef = doc(database, DATABASE_FOLDER_NAME, userUID);
+  const [userBio, setUserBio] = useState(userData?.Bio);
+  const [userSelectedCategories, setUserSelectedCategories] = useState(
+    userData?.selectedCategories ?? []
+  );
+  const [availableCategories, setAvailableCategories] = useState([]);
+
+  useEffect(() => {
+    const loadCategories = async () => {
+      const fetchedCategories = await fetchCategoryNames();
+      setAvailableCategories(fetchedCategories);
+    };
+
+    loadCategories();
+  }, []);
+  const toggleEditProfileModal = () => {
+    setIsEditProfileModalShown(!isEditProfileModalShown);
+  };
+
+  const handleBioChange = (e) => {
+    setUserBio(e.target.value);
+  };
```

Comment:
Reminder to run formatter before submission. Prefer an empty line between functions.